pallets/msa/src/tests.rs

```diff
@@ -955,3 +982,101 @@ pub fn delegation_expired() {
 		);
 	})
 }
+
+#[test]
+fn signed_extension_revoke_msa_delegation_by_delegator() {
+	new_test_ext().execute_with(|| {
+		let (key_pair, _) = sr25519::Pair::generate();
+		let provider_account = key_pair.public();
+
+		let add_provider_payload = AddProvider { authorized_msa_id: 1, permission: 0 };
+		let encode_add_provider_data = wrap_binary_data(add_provider_payload.encode());
+
+		let signature: MultiSignature = key_pair.sign(&encode_add_provider_data).into();
+
+		assert_ok!(Msa::create(test_origin_signed(1)));
+		assert_ok!(Msa::create(Origin::signed(provider_account.into())));
+		assert_ok!(Msa::add_provider_to_msa(
+			test_origin_signed(1),
+			provider_account.into(),
+			signature,
+			add_provider_payload
+		));
+
+		let provider: MessageSourceId = 2;
+		let call_revoke_delegation: &<Test as frame_system::Config>::Call =
+			&Call::Msa(MsaCall::revoke_msa_delegation_by_delegator { provider_msa_id: provider });
+		let info = DispatchInfo::default();
+		let len = 0_usize;
+		let result = CheckProviderRevocation::<Test>::new().validate(
+			&test_public(1),
+			call_revoke_delegation,
+			&info,
+			len,
+		);
+		assert_ok!(result);
+	});
+}
+
+#[test]
+fn signed_extension_validation_failure_on_revoked() {
+	new_test_ext().execute_with(|| {
+		let (key_pair, _) = sr25519::Pair::generate();
```

Comment:
Non-blocking: A function that does this setup (create keypairs and providers and then add one as a delegate) instead of duplicating it would be really useful, especially for future tests of this pallet.